### PR TITLE
markEmployees improvements

### PR DESCRIPTION
**Pull request by double-beep**
- Fix typos
- Implement pagination
- Move `$icon` declaration and CSS style out of function
- Don't remove the logo when DOM updates (new comment added or review post appears)

### CODE REVIEW

File: sox.common.js [hunk 1]
```diff
@@ -197,6 +198,7 @@
       if (filter) queryParams.push(`filter=${filter}`);
       if (order) queryParams.push(`order=${order}`);
       if (limit) queryParams.push(`pagesize=${limit}`);
+      if (page) queryParams.push(`page=${page}`)
```
Comment (shu8) [inline sox.common.js]: Tiny nitpick, sorry: can you add a semicolon at the end of this line? (I know it's not necessary but it's the style the rest of SOX uses!)

File: sox.features.js [hunk 2]
```diff
@@ -35,45 +35,57 @@
     },
 
     markEmployees: function () {
-      // Description: Adds an Stack Overflow logo next to users that *ARE* a Stack Overflow Employee
+      // Description: Adds the Stack Exchange logo next to users that *ARE* Stack Exchange employees
```
Comment (shu8) [inline sox.features.js]: Do you mind also updating this (the fact that it's the SE logo instead of SO) in the sox.features.json file?

File: sox.features.js [hunk 3]
```diff
@@ -35,45 +35,57 @@
...
 
-      for (let i = 0; i < anchors.length; i++) {
-        const href = anchors[i].href;
-        const idMatch = href.match(/(\d+)/);
-        if (idMatch && ids.indexOf(idMatch[1]) === -1) ids.push(idMatch[1]);
-      }
-      sox.debug('markEmployees user IDs', ids);
+      function getIds() {
+        const anchors = [...document.querySelectorAll('.comment a, .deleted-answer-info a, .employee-name a, .user-details a, .question-summary .started a')].filter(el => {
+          return !el.parentElement.classList.contains('user-gravatar32') && $(el).find('.sox-markEmployees-logo') && el.href && el.href.contains('/users/');
```
Comment (shu8) [inline sox.features.js]: Should this addition be `!$(el).find('.sox-markEmployees-logo').length` instead, so we exclude anchors with the logo already added?

File: sox.features.js [hunk 4]
```diff
@@ -35,45 +35,57 @@
...
-            }).append($icon.clone()));
-          });
+        for (let i = 0; i < anchors.length; i++) {
+          const idMatch = anchors[i].href.match(/(\d+)/);
+          if (idMatch && ids.indexOf(idMatch[1]) === -1) ids.push(idMatch[1]);
         }
-      });
+        sox.debug('markEmployees user IDs', ids);
+
+        for (let i = 1; i < Math.trunc(ids.length / 100) + 2; i++) {
```
Comment (shu8) [inline sox.features.js]: If `ids.length` is `100` here, will the loop `i < Math.trunc(ids.length / 100) + 2` condition be 3 instead of 2 which is what we'd actually want?

File: sox.features.js [hunk 5]
```diff
@@ -35,45 +35,57 @@
...
+      }
+
+      function apiCall(page, ids, anchors) {
+        sox.helpers.getFromAPI({
+          endpoint: 'users',
+          ids,
+          sitename: sox.site.url,
+          filter: '!*MxJcsv91Tcz6yRH',
+          limit: 100,
+          page: page,
```
Comment (shu8) [inline sox.features.js]: You could shorten this to just `page,`

File: sox.features.js [hunk 6]
```diff
@@ -35,45 +35,57 @@
...
+          featureId: 'markEmployees',
+          cacheDuration: 60 * 24, // Cache for 24 hours (in minutes)
+        }, items => {
+          sox.debug('markEmployees returned data', items);
+          for (let i = 0; i < items.length; i++) {
+            const userId = items[i].user_id;
+            if (!items[i].is_employee) continue;
+
+            anchors.filter(el => el.href.contains(`/users/${userId}/`)).forEach(el => {
+              if (!$(el).find('.sox-markEmployees-logo').length) {
```
Comment (shu8) [inline sox.features.js]: Do we need this check again if it was included in the `.filter` call when we populated the `anchors` array?

File: sox.features.js (hunk 4 above)
Comment (double-beep) [reply sox.features.js]: A simple `for (let i = 1; i < Math.trunc(100 / 100) + 2; i++) { console.log(i); }` says `1` and `2` - note that `i` isn't `0`, but `1`.

**Comment by double-beep** [comment]
I'm ready, implemented all suggestions. Because there are many commits, I think it's better to squash & merge this one.

File: sox.features.js (hunk 4 above)
Comment (shu8) [reply sox.features.js]: If we've got a page length of 100 though, do we not just want page `1` and not `2`?

Comment (double-beep) [reply sox.features.js]: Most probably /me who thought 100 ids need 2 API calls. Will think of it tomorrow - if you have a suggestion, can you please add a comment?

Comment (shu8) [reply sox.features.js]: I think:
```
for (let i = 0; i < Math.ceil(ids.length / 100); i++) {
    apiCall(i + 1, ids.slice(i * 100, (i * 100) + 100), anchors);
}
```
should work fine

Comment (double-beep) [reply sox.features.js]: My idea was:

    for (let i = 1; i < Math.round(ids.length * 100) / 10000 + 1; i++) {
      apiCall(i, ids.slice(i * 100 - 100, i * 100), anchors);
     }

Comment (shu8) [reply sox.features.js]: I think the `ceil` version would probably be more readable and easier to understand because it doesn't have the division by 10000 and reads a bit more naturally.

You could still start the loop at index 1 thought if you would prefer that!

**Comment by shu8** [comment]
Thanks for the changes @double-beep! Squashed and merged! :)
